posts/2021-03-25-Rust-1.51.0.md

```diff
@@ -0,0 +1,179 @@
+
+While this can lead to a compiling some crates more than once, this should provide a much more intuitive development experience when using features with cargo. If you'd like to know more, you can also read the ["Feature Resolver"][feature-resolver@2.0] section in the Cargo Book for more information. We'd like to thank the cargo team and everyone involved for all their hard work in designing and implementing this new feature (resolver)!
+
+[feature-resolver@2.0]: https://doc.rust-lang.org/nightly/cargo/reference/features.html#feature-resolver-version-2
+
+### Splitting Debug Information
+While not often highlighted in the release, the Rust teams are constantly working on improving compile Rust's compile times, and this release marks one biggest improvements in a long time for Rust on macOS platforms. Debug information is code that maps the binary code back to your source code, so that the program can give you more information about what when wrong at run time. In macOS this is done with a tool called `dsymutil` which collects all the debug information into a single `.dSYM` file.
+
+This tool provides a lot of benefits, however one of the drawbacks has been that `dsymutil` is not really compatible with incremental recompilation, so even when you make a small change such as a function name, `dsymutil` will need to run over the entire final binary to produce the final `.dSYM` folder. This can add a lot to the build time, especially for larger projects, however this has been a necessary step as without it Rust wouldn't be able to parse debug info on macOS.
+
+However with the implementation of the Rust based `backtrace` crate backend, we're now able to parse debuginfo even without running `dsymutil`. This can significantly speed up builds that include debuginfo. We haven't ran extensive benchmarks but we've seen a lot of anecdata of people's builds building a lot faster on macOS with this behaviour.
+
+You can enable this new behaviour by setting. the `-Csplit-debuginfo=unpacked` flag when running `rustc`, or by setting the [`split-debuginfo`] `[profile]` option to `unpacked` in Cargo.
+
+```toml
+[profile.dev]
+split-debuginfo = "unpacked"
+```
+
+[`split-debuginfo`]: https://doc.rust-lang.org/nightly/cargo/reference/profiles.html#split-debuginfo
+
+### Stabilized APIs
+
+In total, this release saw the stabilisation of 18 new methods for various types like `slice` and `Peekable`. One notable addition is the stabilisation of `ptr::add_of!` and `ptr::addr_of_mut!`, which allow you to create unaligned references to data	, previously this wasn't possible because Rust requires `&/&mut` to be aligned and point to initialized data, and `&addr as *const _` would cause undefined behaviour `&addr` needs to be aligned. These two macros now let you safely create unaligned references.
```

Comment:
```suggestion
In total, this release saw the stabilisation of 18 new methods for various types like `slice` and `Peekable`. One notable addition is the stabilisation of `ptr::addr_of!` and `ptr::addr_of_mut!`, which allow you to create unaligned references to data	, previously this wasn't possible because Rust requires `&/&mut` to be aligned and point to initialized data, and `&addr as *const _` would cause undefined behaviour `&addr` needs to be aligned. These two macros now let you safely create unaligned references.
```

Reply:
```suggestion
In total, this release saw the stabilisation of 18 new methods for various types like `slice` and `Peekable`. One notable addition is the stabilisation of `ptr::addr_of!` and `ptr::addr_of_mut!`, which allow you to create unaligned references to data	, previously this wasn't possible because Rust requires `&/&mut` to be aligned and point to initialized data, and `&addr as *const _` would cause undefined behaviour as `&addr` needs to be aligned. These two macros now let you safely create unaligned references.
```